rust/hyperlane-base/src/contract_sync/cursors/rate_limited.rs

```diff
@@ -146,10 +78,10 @@ pub enum SyncDirection {
 /// queried is and also handling rate limiting. Rate limiting is automatically
 /// performed by `next_action`.
 pub(crate) struct RateLimitedContractSyncCursor<T> {
+    // TODO: Sequence information is no longer required, so just use Indexer
```

Comment:
iiuc we can make this move now to `indexer: Arc<dyn Indexer<T>>,`? and to get the tip everywhere we use `let tip = this.indexer.get_finalized_block_number().await?;`